worker.py

```diff
@@ -82,6 +75,20 @@ def notify_task_completion(self, task):
         self.add_random_delay()
         pass
 
+    def get_result_from_worker(self, future: Future) -> bytes:
+        if (future.hostName == self.worker_host and future.port == self.worker_port):
```

Comment:
doesnt future have a worker id field? on what worker has the future been scheduled to?

Reply:
No future only has hostname and port to fetch results from

Reply:
it would be good to have a worker id instead for privacy reasons.

Reply:
Worker has no knowledge on how to fetch hostname and port from workerId of another worker unless it polls the scheduler, which is an extra network call we can avoid.
Also, Why is privacy a concern here?